core/src/main/scala/akka/kafka/scaladsl/MetadataClient.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+import akka.actor.{ActorRef, ActorSystem}
+import akka.kafka.{ConsumerSettings, KafkaConsumerActor}
+import akka.kafka.Metadata.{BeginningOffsets, GetBeginningOffsets}
+import akka.pattern.ask
+import akka.util.Timeout
+import org.apache.kafka.common.TopicPartition
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object MetadataClient {
+
+  private val consumerActors = scala.collection.mutable.Map[ConsumerSettings[Any, Any], ActorRef]()
+
+  def getBeginningOffsets[K, V](
+      consumerSettings: ConsumerSettings[K, V],
+      partitions: Set[TopicPartition],
+      timeout: Timeout
+  )(implicit system: ActorSystem, ec: ExecutionContext): Future[Map[TopicPartition, Long]] = {
```

Comment:
The `ask` and `map` operations are safe to be executed on the ActorSystem's dispatcher, you can get away without the separate `ExecutionContext` here.